pkg/controlplane/instance.go

```diff
@@ -674,9 +676,14 @@ type RESTStorageProvider interface {
 // InstallAPIs will install the APIs for the restStorageProviders if they are enabled.
 func (m *Instance) InstallAPIs(apiResourceConfigSource serverstorage.APIResourceConfigSource, restOptionsGetter generic.RESTOptionsGetter, restStorageProviders ...RESTStorageProvider) error {
 	nonLegacy := []*genericapiserver.APIGroupInfo{}
-
+	var resourceExpirationEvaluator genericapiserver.ResourceExpirationEvaluator
+	var err error
 	// used later in the loop to filter the served resource by those that have expired.
-	resourceExpirationEvaluator, err := genericapiserver.NewResourceExpirationEvaluator(*m.GenericAPIServer.Version)
+	if utilfeature.DefaultFeatureGate.Enabled(apiserverfeatures.EmulationVersion) {
+		resourceExpirationEvaluator, err = genericapiserver.NewResourceExpirationEvaluator(serverversion.Effective.EmulationVersion())
+	} else {
+		resourceExpirationEvaluator, err = genericapiserver.NewResourceExpirationEvaluator(apimachineryversion.MustParse(m.GenericAPIServer.Version.String()))
+	}
```

Comment:
Do we not set emulation version always? Just wondering if we can possibly get rid of this conditional..